### PR TITLE
MAINT: stats.triang edge case fixes #6036

### LINKED ISSUE #6036: Incorrect edge case in scipy.stats.triang.pdf
When `c = 1`, the pdf of the triangular distribution should just be `f(x) = 2x`.  But as of version 0.17, [the pdf is computed using the formula](https://github.com/scipy/scipy/blob/v0.17.0/scipy/stats/_continuous_distns.py#L4121),

```
where(x < c, 2*x/c, 2*(1-x)/(1-c))
```

which incorrectly returns `nan`, rather than 2, in the case `x = 1`, `c = 1`.

### PR DESCRIPTION
**Pull request by andyfaff**
Fixes #6036

### CODE REVIEW

**Comment by ilayn** [comment]
I've added the issue number to the PR message otherwise they are not linked.

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -5240,10 +5240,28 @@ def _argcheck(self, c):
         return (c >= 0) & (c <= 1)
 
     def _pdf(self, x, c):
-        return np.where(x < c, 2*x/c, 2*(1-x)/(1-c))
+        r = _lazyselect([c == 0,
+                         c == 1,
+                         x < c,
+                         x >= c],
+                        [lambda x, c: 2 - 2 * x,
```
Comment (rgommers) [inline scipy/stats/_continuous_distns.py]: style nitpick: the preferred style (and current PEP8 recommendation) is to group things a bit more: `2 - 2*x`, `2 * (1-x) / (1-c)`, etc.

File: scipy/stats/_continuous_distns.py [hunk 2]
```diff
@@ -5240,10 +5240,28 @@ def _argcheck(self, c):
...
     def _cdf(self, x, c):
-        return np.where(x < c, x*x/c, (x*x-2*x+c)/(c-1))
+        r = _lazyselect([c == 0,
+                         c == 1,
+                         x < c,
+                         x >= c],
+                        [lambda x, c: 2 * x - x * x,
+                         lambda x, c: x * x,
+                         lambda x, c: x * x / c,
+                         lambda x, c: (x*x-2*x+c)/(c-1)],
```
Comment (rgommers) [inline scipy/stats/_continuous_distns.py]: suggest `(x*x - 2*x + c) / (c-1)`

**Comment by rgommers** [comment]
Changed behavior LGTM. Would be good to squash the commits after pushing the last style fix..

**Comment by andyfaff** [comment]
ACK, PEP8 changes made and squashed.

**Comment by rgommers** [comment]
Merged, thanks @andyfaff

**Pull request merged**